lib/telemetry_registry.ex

```diff
@@ -150,65 +150,86 @@ defmodule TelemetryRegistry do
   """
   @type spannable_event() :: :telemetry_registry.spannable_event()
 
-
   defmacro __using__(_opts) do
     quote do
-      import unquote(__MODULE__), only: [format_telemetry_docs: 1, telemetry_event: 1]
+      import unquote(__MODULE__), only: [telemetry_docs: 0, telemetry_event: 1]
 
       Module.register_attribute(__MODULE__, :telemetry_event,
         accumulate: true,
         persist: true
       )
-
-      Module.register_attribute(__MODULE__, :telemetry_docs,
-        accumulate: true,
-        persist: true
-      )
     end
   end
 
   @doc """
-  Declares a telemetry event. Accepts a telemetry event definition or telemetry event name. A full
-  definition is required for complete documentation and is highly encouraged.
+  Declares a telemetry event. Accepts a telemetry event definition `t:event_definition/0` or telemetry
+  event name `t:event/0`. A full definition is required for complete documentation and is highly encouraged.
   """
   defmacro telemetry_event(event) do
-    docstring =
-      case event do
-        event when is_list(event) ->
-          """
-          * `#{inspect(event)}`
-
-          """
-
-        {:%{}, _, event} ->
-          """
-          * `#{inspect(event[:event])}`
-            * Description: #{event[:description]}
-            * Measurements: `#{event[:measurements]}`
-            * Metadata: `#{event[:metadata]}`
-
-          """
-      end
-
     quote do
       @telemetry_event unquote(event)
-      @telemetry_docs unquote(docstring)
     end
   end
 
   @doc """
-  Formats telemetry definitions declared in the module for inclusion in documentation.
-
-  Example
+  Generates telemetry event documentation formatted in Markdown for use in your documentation.
 
   ```
   ## Telemetry
 
-  #{format_telemetry_docs(@telemetry_docs)}
+  \#{telemetry_docs()}
   ```
   """
-  @spec format_telemetry_docs([String.t()]) :: String.t()
-  def format_telemetry_docs(docs), do: Enum.reverse(docs) |> IO.iodata_to_binary()
+  defmacro telemetry_docs do
```

Comment:
What do you think about my later suggestion of actually passing the module? And we try to load the available events either internally (module attribute) or externally?